RemoteCarControllerClient/app/src/main/java/org/elsys/remote_control_car/MainActivity.java

```diff
@@ -19,4 +37,28 @@ private void enableFullScreenLandscape() {
                              WindowManager.LayoutParams.FLAG_FULLSCREEN);
         setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
     }
+
+    private void initializeButtons() {
+        buttons = new HashMap<String, ImageButton>();
+
+        buttons.put("Forward", (ImageButton) findViewById(R.id.forward_btn));
+        buttons.put("Backward", (ImageButton) findViewById(R.id.backward_btn));
+        buttons.put("Right", (ImageButton) findViewById(R.id.right_btn));
+        buttons.put("Left", (ImageButton) findViewById(R.id.left_btn));
+    }
+
+    private void setListeners() {
+        Iterator<String> iterator = buttons.keySet().iterator();
+
+        while (iterator.hasNext()) {
+            String key = iterator.next();
+            ImageButton button = buttons.get(key);
+
+            if ((key == null) || (button == null)) {
```

Comment:
Relocate setListeners() function to be called from initializeButtons() functions. Remove "if ((key == null) || (button == null))" block and throwing and handling of NullPointerException, because iterator.hasNext() do the check.